src/Adapter/Presenter/Product/ProductLazyArray.php

```diff
@@ -462,6 +500,90 @@ public function getEcotax()
         return null;
     }
 
+    /**
+     * @arrayAccess
+     *
+     * @return string|null
+     */
+    public function getManufacturerName()
+    {
+        if (!isset($this->product['manufacturer_name'])) {
+            // Assign empty value
+            $this->product['manufacturer_name'] = null;
+
+            // If we have manufacturer ID, we will try to load it's name and assign it
+            if (!empty($this->product['id_manufacturer'])) {
+                $manufacturerName = Manufacturer::getNameById((int) $this->product['id_manufacturer']);
+                if (!empty($manufacturerName)) {
+                    $this->product['manufacturer_name'] = $manufacturerName;
+                }
+            }
+        }
+
+        return $this->product['manufacturer_name'];
+    }
```

Comment:
### Suggestion : 
rather improve the `Manufacturer::getNameById` to return a null when the param `$idManufacturer` provided is not valid.

```suggestion
    public function getManufacturerName()
    {
        if (!isset($this->product['manufacturer_name'])) {
            $this->product['manufacturer_name'] = Manufacturer::getNameById((int) $this->product['id_manufacturer']);
        }

        return $this->product['manufacturer_name'];
    }
```